pkg/api/validate/vm.go

```diff
@@ -7,66 +7,97 @@ import (
 	"github.com/Azure/ARO-RP/pkg/api"
 )
 
+var supportedMasterVMSizes = []api.VMSize{
+	// General purpose
+	api.VMSizeStandardD8sV3,
+	api.VMSizeStandardD16sV3,
+	api.VMSizeStandardD32sV3,
+	// Memory optimized
+	api.VMSizeStandardE64iV3,
+	api.VMSizeStandardE64isV3,
+	api.VMSizeStandardE80isV4,
+	api.VMSizeStandardE80idsV4,
+	api.VMSizeStandardE104iV5,
+	api.VMSizeStandardE104isV5,
+	api.VMSizeStandardE104idV5,
+	api.VMSizeStandardE104idsV5,
+	// Compute optimized
+	api.VMSizeStandardF72sV2,
+	// Memory and storage optimized
+	api.VMSizeStandardGS5,
+	api.VMSizeStandardG5,
+	// Memory optimized
+	api.VMSizeStandardM128ms,
+}
+
+var supportedWorkerVMSizes = []api.VMSize{
+	// General purpose
+	api.VMSizeStandardD4asV4,
+	api.VMSizeStandardD8asV4,
+	api.VMSizeStandardD16asV4,
+	api.VMSizeStandardD32asV4,
+	api.VMSizeStandardD4sV3,
+	api.VMSizeStandardD8sV3,
+	api.VMSizeStandardD16sV3,
+	api.VMSizeStandardD32sV3,
+	// Memory optimized
+	api.VMSizeStandardE4sV3,
+	api.VMSizeStandardE8sV3,
+	api.VMSizeStandardE16sV3,
+	api.VMSizeStandardE32sV3,
+	api.VMSizeStandardE64isV3,
+	api.VMSizeStandardE64iV3,
+	api.VMSizeStandardE80isV4,
+	api.VMSizeStandardE80idsV4,
+	api.VMSizeStandardE104iV5,
+	api.VMSizeStandardE104isV5,
+	api.VMSizeStandardE104idV5,
+	api.VMSizeStandardE104idsV5,
+	// Compute optimized
+	api.VMSizeStandardF4sV2,
+	api.VMSizeStandardF8sV2,
+	api.VMSizeStandardF16sV2,
+	api.VMSizeStandardF32sV2,
+	api.VMSizeStandardF72sV2,
+	// Memory and storage optimized
+	api.VMSizeStandardG5,
+	api.VMSizeStandardGS5,
+	// Memory optimized
+	api.VMSizeStandardM128ms,
+	// Storage optimized
+	api.VMSizeStandardL4s,
+	api.VMSizeStandardL8s,
+	api.VMSizeStandardL16s,
+	api.VMSizeStandardL32s,
+	api.VMSizeStandardL8sV2,
+	api.VMSizeStandardL16sV2,
+	api.VMSizeStandardL32sV2,
+	api.VMSizeStandardL48sV2,
+	api.VMSizeStandardL64sV2,
+}
+
```

Comment:
Why not use a `map[api.VMSize]struct{}` or `map[api.VMSize]bool` instead of doing a lookup with a loop?

Reply:
Good call, I went with a `map[api.VMSize]bool`